compiler/testData/diagnostics/tests/scopes/initializerScopeOfExtensionProperty.fir.kt

```diff
@@ -24,8 +24,8 @@ class C {
     inner class D {}
 }
 
-val C.foo : C.D = <!UNRESOLVED_REFERENCE!>D<!>()
+val C.foo : C.D = <!EXTENSION_PROPERTY_WITH_BACKING_FIELD!><!UNRESOLVED_REFERENCE!>D<!>()<!>
 
-val C.bar : C.D = C().D()
+val C.bar : C.D = C().<!EXTENSION_PROPERTY_WITH_BACKING_FIELD!>D()<!>
```

Comment:
This should be reported on `C().D()`, not `D()`. I explained why, and suggested a couple way to fix: https://youtrack.jetbrains.com/issue/KT-42591 PTAL that issue and leave comments about what option you prefer, there or here.

Reply:
@mglukhikh Could you take a look at that issue about wrong source of qualified access expression?

Reply:
Yes, I'm planning to look at that next week.